sklearn/linear_model/tests/test_ridge.py

```diff
@@ -421,6 +422,10 @@ def _make_sparse_offset_regression(
     X[~mask] = 0.0
     removed_X[mask] = 0.0
     y -= removed_X.dot(c)
+    if positive:
+        c_diff = c + np.random.RandomState(random_state).uniform(1, 10, c.shape)
+        y += X.dot(c_diff)
+        c += c_diff
```

Comment:
```suggestion
        y += X.dot(np.abs(c) - c)
        c = np.abs(c)
```

Reply:
I'm sorry. we have to use abs of coefficients. FIxed.
But I left the logic to add random positive noise to avoid negative coefficient solution.

Reply:
`y` was originally constructed as `X@c + noise`. What I wanted to achieve is `y = X@np.abs(c) + noise`. I don't think we need the positive random noise. If there are numerical issues, we can add a small number, e.g.
```
y += X@(np.abs(c) + 0.1 - c)
c = np.abs(c) + 0.1
```

Reply:
Fixed:
``` python
y += X.dot(np.abs(c) + 1 - c)
c = np.abs(c) + 1
```